sync/task_pool.go

```diff
@@ -158,14 +159,54 @@ func (p *TaskPool) run(id int, q chan task) error {
 	}
 }
 
-// add task
+// AddTask wait idle worker add task
 func (p *TaskPool) AddTask(t task) {
+	idx := atomic.AddUint32(&p.idx, 1)
+	id := idx % uint32(p.tQNumber)
+
+	select {
+	case <-p.done:
+		return
+	case p.qArray[id] <- t:
+	}
+}
+
+// AddTaskAlways add task to queues or do it immediately
+func (p *TaskPool) AddTaskAlways(t task) {
 	id := atomic.AddUint32(&p.idx, 1) % uint32(p.tQNumber)
 
 	select {
 	case <-p.done:
 		return
 	case p.qArray[id] <- t:
+		return
+	default:
+		go t()
```

Comment:
pls do not use 'go'. use 'gxruntime.GoSafely' instead.

Reply:
updated.

and please review func p.safeRun and goSafely https://github.com/dubbogo/gost/pull/25/files#diff-e3fb3a64c85404e71d3fd89a55036c36R134